app/models/integration_mappings/base_mapping.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module IntegrationMappings
+  class BaseMapping < ApplicationRecord
+    include PaperTrailTraceable
+    include SettingsStorable
+
+    self.table_name = 'integration_mappings'
+
+    belongs_to :integration, class_name: 'Integrations::BaseIntegration'
+    belongs_to :mappable, polymorphic: true
+
+    MAPPABLE_TYPES = descendants.map do |descendant|
```

Comment:
Maybe we can move here: `MAPPABLE_TYPES = %i[AddOn BillableMetric].freeze`
When I think again I am not sure it will change with different providers... wdyt?

Reply:
Yes, moving it here.